### PR TITLE
Elijah login system

### PR DESCRIPTION
**Pull request by ezwendel**
Made a backend and frontend (needs work) for the login system.

Test: Can go to the url http://localhost:3000/login and enter the username johnIsCool and password reallyCoolGuy1 and login with no problems. If not using that account, there will be errors shown to the user (run "npm i" before testing). 

It should redirect to /home which is an error page.

### CODE REVIEW

File: tasks/seed.js [hunk 1]
```diff
@@ -1 +1,53 @@
 // This file is to seed the database with predetermined values when being installed on a new machine
+
+const express = require('express')
+const router = express.Router()
+const dbConnection = require('../config/mongoConnection');
+const data = require('../data');
+const userData = data.users
+
+// let usersSchema = mongoose.Schema({
```
Comment (cs-paz) [inline tasks/seed.js]: Remove this

File: routes/login.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    if (!comparison) {
+      console.log('incorrect password')
+      return res.status(400).render('login', {
+        title: 'Log In',
+        error: 'Username and/or password is incorrect.'
+      });
+    }
+
+    req.session.user = user; // have to change to make cookie secure
+    console.log('here')
```
Comment (cs-paz) [inline routes/login.js]: remove this

File: routes/login.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  try {
+    const users = await userData.getAllUsers();
+
+    let user = null;
+    username = username.toLowerCase();
+    console.log(users)
+    for (i of users) { // search through all users for username
+      console.log(i)
```
Comment (cs-paz) [inline routes/login.js]: remove this

File: routes/login.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      console.log(i.Username)
```
Comment (cs-paz) [inline routes/login.js]: remove this

File: routes/login.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+/* GET admin page */
+router.get('/', async (req, res, next) => {
+  res.render('login', {
+    title: 'Log In'
+  });
+});
+
+router.post('/submitLogin/', async (req, res) => {
+  console.log(req.body)
```
Comment (cs-paz) [inline routes/login.js]: remove this

File: routes/login.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  let username = req.body.username
+  const password = req.body.password
+  console.log(username)
```
Comment (cs-paz) [inline routes/login.js]: remove this

File: routes/login.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  console.log(password)
```
Comment (cs-paz) [inline routes/login.js]: remove this

File: data/users.js [hunk 8]
```diff
@@ -0,0 +1,107 @@
+const { ObjectId } = require('mongodb');
+const { get } = require('mongoose');
+const { users } = require('../config/mongoCollections');
+const bcrypt = require('bcryptjs')
+
+
+// let usersSchema = mongoose.Schema({
```
Comment (cs-paz) [inline data/users.js]: remove this

**Review by cs-paz — CHANGES_REQUESTED** [review]

**Review by cs-paz — APPROVED** [review]

**Review by ashnarazdan123 — COMMENTED** [review]
Started at http://localhost:3000/login and used username johnIsCool and password reallyCoolGuy1. Brought to message "error not found" at http://localhost:3000/home

If a different password or username entered "Username and/or password is incorrect" entered so that looks good. 

Something to note: if you remove the /home from url it brings you back to the http://localhost:3000 page with calendar without asking for another login.

File: data/users.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    newUser.Role = formRequestBody.Role;
+    console.log(newUser);
+
+    try {
+        User.create(newUser);
+    } catch (e) {
+        console.log(e);
+    }
+
+    // no clue what this does
```
Comment (krixstin) [inline data/users.js:36]: I believe **.find()** method instantiates query, so it should let us use some searching parameters in {}, for like sorting or filtering later when needed

--->example from Geeksforgeeks 

// Only one parameter [query/condition]
// Find all documents that matches the
// condition name='Punit'
User.find(**{ name: 'Punit'}**, function (err, docs) {
    if (err){
        console.log(err);
    }
    else{
        console.log("First function call : ", docs);
    }
});

**Review by krixstin — APPROVED** [review]

**Pull request merged**